File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        String parsed = parseString(s);
        if(parsed.isEmpty()) {
            return true;
        }

        int left = 0;
        int right = parsed.length() - 1;

        while(left < right) {
            if(parsed.charAt(left) == parsed.charAt(right)) {
                left++;
                right--;
            }
            else {
                return false;
            }
        }

        return true;
    }

    private String parseString(String s) {
        StringBuilder sb = new StringBuilder();
        
        for(char c : s.toCharArray()) {
            if(Character.isLetterOrDigit(c)) {
                sb.append(Character.toLowerCase(c));
            }
        }

        return sb.toString();

    }
}

File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.ts
function isPalindrome(s: string): boolean {

    // String parsed = s.toLowerCase().replaceAll("[^a-z0-9]", "");
    let parsed: string = s.toLowerCase().replaceAll(/[^a-z0-9]/g, "");
    if (parsed.length === 0) {
        return true;
    }

    let left: number = 0;
    let right: number = parsed.length - 1;

    while (left < right) {
        if (parsed[left] == parsed[right]) {
            left++;
            right--;
        }
        else {
            return false;
        }
    }

    return true;
};